apps/web/src/pages/integrations/components/multi-provider/UpdateProviderSidebar.tsx

```diff
@@ -195,20 +197,27 @@ export function UpdateProviderSidebar({
 
     const hasConditionsAndIsPrimary = hasUpdatedConditions && primary && dirtyFields.conditions;
 
-    if (
-      (hasNoConditions && isChangedToActive) ||
-      isChangedToInactiveAndIsPrimary ||
-      isPrimaryAndHasConditionsApplied ||
-      hasConditionsAndIsPrimary
-    ) {
-      openSelectPrimaryIntegrationModal({
+    const selectPrimaryIntegration = () => {
+      return openSelectPrimaryIntegrationModal({
         environmentId: selectedProvider?.environmentId,
         channelType: selectedProvider?.channel,
         exclude: !isActive || hasConditionsAndIsPrimary ? (el) => el._id === selectedProvider.integrationId : undefined,
         onClose: () => {
           updateIntegration(data);
         },
       });
+    };
+
+    if (isChangedToInactiveAndIsPrimary) {
+      openDisablePrimaryStatusModal({
+        onConfirm: () => selectPrimaryIntegration(),
+      });
+
+      return;
+    }
+
+    if ((hasNoConditions && isChangedToActive) || isPrimaryAndHasConditionsApplied || hasConditionsAndIsPrimary) {
+      selectPrimaryIntegration();
 
       return;
     }
```

Comment:
simply split previous one big `if` statement to two smaller, because I need to show 2 modals when `isChangedToInactiveAndIsPrimary` is `true`: warning + primary integration select.

When user clicks `Cancel` in **warning modal**, nothing happens - it closes modal and does nothing (no network calls etc).

When user clicks `Disable instance` it just shows select primary integration modal, updates integration on close as expected.